src/Xabe.FFmpeg/Conversion/FFmpegWrapper.cs

```diff
@@ -43,12 +49,17 @@ internal class FFmpegWrapper : FFmpeg
             return Task.Factory.StartNew(() =>
             {
                 _outputLog = new List<string>();
-                var process = RunProcess(args, FFmpegPath, priority, true, false, true);
+                bool pipedOutput = OnVideoDataReceived != null;
+                var process = RunProcess(args, FFmpegPath, priority, true, pipedOutput, true);
                 var processId = process.Id;
                 using (process)
                 {
                     process.ErrorDataReceived += (sender, e) => ProcessOutputData(e, args, processId);
                     process.BeginErrorReadLine();
+                    if (pipedOutput)
+                    {
+                        Task.Run(() => ProcessVideoData(process));
```

Comment:
Pass here cancellation token

Reply:
I've passed the cancellation token in my new commit. 

Perhaps do you think it would be better to convert `ProcessVideoData` into a lambda? I had to pass the token into both the task and the action.

Reply:
It's fine as it is. Thats internal method so change won't require any docs update nor anything